File: Python/1357. Apply Discount Every n Orders.py
class Cashier:

    def __init__(self, n: int, discount: int, products: List[int], prices: List[int]):
        self.n = n
        self.discount = ((100 - discount) / 100)
        self.customer = 1
        self.goods = {x: y for x, y in zip(products, prices)}

    def getBill(self, product: List[int], amount: List[int]) -> float:
        price = 0

        for p, a in zip(product, amount): price += self.goods[p] * a
        
        if self.customer % self.n == 0: price = price * self.discount
        
        self.customer += 1

        return price


# Your Cashier object will be instantiated and called as such:
# obj = Cashier(n, discount, products, prices)
# param_1 = obj.getBill(product,amount)

File: Java/1357. Apply Discount Every n Orders.java
class Cashier {
    public int n, customer;
    public double discount, price;
    public HashMap<Integer, Integer> goods = new HashMap<>();

    public Cashier(int n, int discount, int[] products, int[] prices) {
        this.n = n;
        this.discount = ((100d - discount) / 100d);
        this.customer = 1;
        for(int i = 0; i < products.length; i++) goods.put(products[i], prices[i]);
    }
    
    public double getBill(int[] product, int[] amount) {
        price = 0.0d;

        for(int i = 0; i < product.length; i++) price += goods.get(product[i]) * amount[i];

        if(customer % n == 0) price *= discount;

        customer++;

        return (double)price;
    }
}

/**
 * Your Cashier object will be instantiated and called as such:
 * Cashier obj = new Cashier(n, discount, products, prices);
 * double param_1 = obj.getBill(product,amount);
 */
